File: api/Avancira.Application/Identity/Users/Constants/PasswordRules.cs
namespace Avancira.Application.Identity.Users.Constants;

/// <summary>
/// Password complexity rules shared with the frontend.
/// Keep in sync with Frontend.Angular/src/app/validators/password-rules.ts
/// </summary>
public static class PasswordRules
{
    public const int MinLength = 8;
    public const string UppercasePattern = "[A-Z]";
    public const string LowercasePattern = "[a-z]";
    public const string DigitPattern = "[0-9]";
    public const string NonAlphanumericPattern = "[^a-zA-Z0-9]";
}

File: Frontend.Angular/src/app/validators/password-rules.ts
// Shared password rules used across the Angular app.
// Keep in sync with api/Avancira.Application/Identity/Users/Constants/PasswordRules.cs

export const MIN_PASSWORD_LENGTH = 8;
export const UPPERCASE_PATTERN = /[A-Z]/;
export const LOWERCASE_PATTERN = /[a-z]/;
export const DIGIT_PATTERN = /[0-9]/;
export const SYMBOL_PATTERN = /[^a-zA-Z0-9]/;
